include/vcpkg/base/util.h

```diff
@@ -130,12 +130,57 @@ namespace vcpkg::Util
         return ret;
     }
 
+    template<class Container, class T>
+    void erase_remove(Container& cont, const T& el)
+    {
+        cont.erase(std::remove(cont.begin(), cont.end(), el), cont.end());
+    }
     template<class Container, class Pred>
     void erase_remove_if(Container& cont, Pred pred)
     {
         cont.erase(std::remove_if(cont.begin(), cont.end(), pred), cont.end());
     }
 
+    // 0th is the first occurence
+    // so find_nth({1, 2, 1, 3, 1, 4}, 1, 2)
+    // returns the 1 before the 4
+    template <class R, class V>
+    auto find_nth(R&& r, const V& v, size_t n)
+    {
+        using std::begin;
+        using std::end;
+
+        auto last = end(r);
+        auto current = std::find(begin(r), last, v);
+
+        for (size_t i = 0; i < n && current != last; ++i)
+        {
+            current = std::find(current + 1, last, v);
+        }
+
+        return current;
+    }
+
+    // 0th is the last occurence
+    // so find_nth({1, 2, 1, 3, 1, 4}, 1, 2)
+    // returns the 1 before the 2
+    template <class R, class V>
+    auto find_nth_from_last(R&& r, const V& v, size_t n)
```

Comment:
```suggestion
    auto find_nth_from_last(R& r, const V& v, size_t n)
```

?

You never want the caller to move their range into this since you're returning an iterator into that.

Reply:
Also, I think you need to better consider what happens when the Nth occurrence isn't found; I think you really want that to return end but I think this returns begin instead?